client/src/services/LeaderboardService.js

```diff
@@ -0,0 +1,66 @@
+import Api from '@/services/Api'
+import GameLogsService from '@/services/GameLogsService'
+
+function getNemesis(logs) {
+  return null
+}
+
+function getConquerer(logs) {
+  return null
+}
+
+function updateUserLeaderboardData(record, id) {
+  logs = GameLogsService.getPlayerGames(id)
+  record.nemesis = getNemesis(logs)
+  record.conquerer = getConquerer(logs)
+  return record
+}
+
+function getPlayerLeaderboard(params) {
+  return Api().get('leaderboard/player/' + params.id)
+}
+
+function fetchLeaderboard() {
+  return Api().get('leaderboard')
+}
+
+function createPlayerLeaderboard(params) {
+  return Api().post('leaderboard/player/' + params.id)
+}
+
+function updateLeaderboards(battle) {
+  getPlayerLeaderboard({id: battle.playerOne}).then(function(data) {
+    playerOneRecord = data.data
+  })
+
+  getPlayerLeaderboard({id: battle.playerTwo}).then(function(data) {
+    playerTwoRecord = data.data
+  })
```

Comment:
What would be the best way to resolve these requests before moving on?

Reply:
This file is relatively ugly btw, since I am trying different things to best update leaderboard after a game is played